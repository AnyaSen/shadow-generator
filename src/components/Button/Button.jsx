import React from "react";

export default function Button() {
  return (
    <div>
      <h1>Button</h1>
    </div>
  );
}
