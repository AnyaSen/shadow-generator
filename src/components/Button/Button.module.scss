@import "../../assets/scss/mixins";
@import "../../assets/scss/variables";

.Button {
  @include button($color-rose, $color-mulberry);
  width: fit-content;
}
