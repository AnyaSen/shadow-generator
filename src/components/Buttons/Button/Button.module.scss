@import "../../../assets/scss/mixins";
@import "../../../assets/scss/variables";

.Button {
  @include button($color-rose, $color-mulberry, $color-white);
  width: fit-content;
}
