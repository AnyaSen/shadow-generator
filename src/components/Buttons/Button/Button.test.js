import React from "react";
import { render, cleanup } from "@testing-library/react";

import Button from "./Button";

describe("<PrimaryButton />", () => {
  afterEach(cleanup);

  test("Should have correct text.", () => {
    const text = "Generate code";

    const { getByText, container } = render(<Button text={text} />);

    const button = getByText(text);

    expect(button).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
