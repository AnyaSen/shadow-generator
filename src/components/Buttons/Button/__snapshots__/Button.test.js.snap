// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> Should have correct text. 1`] = `
<button
  class="Button"
>
  Generate code
</button>
`;
